inspectit-oce-core/src/test/java/rocks/inspectit/oce/core/SpringTestBase.java

```diff
@@ -0,0 +1,132 @@
+import java.util.stream.Collectors;
+
+import static org.mockito.Mockito.*;
+
+/**
+ * Base class for all tests.
+ */
+@ExtendWith(SpringExtension.class)
+@ContextConfiguration(classes = SpringConfiguration.class, initializers = SpringTestBase.TestContextInitializer.class)
+@TestPropertySource(properties = {
+        "inspectit.config.file-based.watch=false"
+})
+public class SpringTestBase {
+
+    @Autowired
+    private TestContextInitializer.TestInspectitEnvironment env;
+
+    private Appender<ILoggingEvent> mockAppender;
+
+
+    /**
+     * Allows to customize properties while the Context is open.
+     * This method is based on {@link InspectitEnvironment#updatePropertySources(Consumer)},
+     * which therefore also triggers {@link rocks.inspectit.oce.core.config.InspectitConfigChangedEvent}s.
+     * <p>
+     * Any test using this method should also hav the {@link org.springframework.test.annotation.DirtiesContext} annotation.
+     *
+     * @param propsCustomizer the lambda for customizing the properties.
+     */
+    public void updateProperties(Consumer<MockPropertySource> propsCustomizer) {
+        env.updatePropertySources((propsList) -> {
+            propsCustomizer.accept(env.mockProperties);
+        });
+    }
+
+    static class TestContextInitializer implements ApplicationContextInitializer<ConfigurableApplicationContext> {
+
+        /**
+         * This "hack" of a nested class has to be used because there is no other way of passing the soures to customizePropertySources
+         * prior to the call of the superconstructor.
+         */
+        List<PropertySource> testPropertySources;
+
+        class TestInspectitEnvironment extends InspectitEnvironment {
+
+            MockPropertySource mockProperties;
+
+
+            public TestInspectitEnvironment(ConfigurableApplicationContext ctx) {
+                super(ctx);
+            }
+
+            @Override
+            protected void customizePropertySources(MutablePropertySources propsList) {
+                mockProperties = new MockPropertySource();
+                propsList.addFirst(mockProperties);
+                testPropertySources.forEach(propsList::addLast);
+                super.customizePropertySources(propsList);
+            }
+        }
+
+        @Override
+        public void initialize(ConfigurableApplicationContext ctx) {
+            ConfigurableEnvironment defaultEnv = ctx.getEnvironment();
+
+            testPropertySources = defaultEnv.getPropertySources().stream()
+                    .filter(ps -> ps.getName() != StandardEnvironment.SYSTEM_ENVIRONMENT_PROPERTY_SOURCE_NAME)
+                    .filter(ps -> ps.getName() != StandardEnvironment.SYSTEM_PROPERTIES_PROPERTY_SOURCE_NAME)
+                    .collect(Collectors.toList());
+
+            InspectitEnvironment env = new TestInspectitEnvironment(ctx);
+            ctx.setEnvironment(env);
+            //TODO: mock Instrumentation class and add to the context
```

Comment:
Would this really be needed? maybe we should only target test with `Instrumentation` for the system tests.. Not sure.. but anyway you can keep the comment..